File: socket/java/TcpEchoClient.java
package socket.java;

public class TcpEchoClient {
    static String HOST = "127.0.0.1";
    static int PORT = 9933;
    public static void main(String[] args) {
        TcpSock s = null;
        try {
            s = new TcpSock(HOST, PORT, true);
            while (true) {
                s.send(s.input("input: ") + "\n");
                System.out.println(s.recvLine());
            }
        } catch (Exception e) {
            e.printStackTrace();
        } finally {
            try {
                if (s != null)
                    s.closeAll();
            } catch (Exception e) {}
        }
    }
}

File: socket/python/tcp_echo_client.py
'''
sock.py example
'''

import sock

HOST = '127.0.0.1'
PORT = 8322


s = sock.sock('tcp', log=True)
s.connect(HOST, PORT)

while True:
    s.send(input('input: ') + '\n') # input() doesn't contain '\n'
    print(s.recv_with_delimiter('\n'))
    #print(s.recv_fixed_length(3))

s.close()
